### ISSUE TITLE
Console not pinging phantom

### ISSUE THREAD

**Issue by hibob5** [jhead/phantom#117]
So as the title states my console isn't pinging phantom. I have made sure the server is on, everything is up to date, and that they are both connected to the internet. Am I missing something?

edit: im on ps4

**Comment by imnotjack91** [comment]
What os are you using to host?

**Comment by imnotjack91** [comment]
Also, phantom won't broadcast if it's not connecting to the server, have you verified it is connecting (using ios or Android version to connect through the proxy?)

**Comment by hibob5** [comment]
Im on windows. And i have checked on mobile before and it works on both it and ps4 it just stops connecting on and off for a few days at a time.

**Comment by azariah001** [comment]
Hello, we're seeing this behaviour on an Xbox as well. The issue is caused by launching phantom, connecting to the server on the Xbox, playing for any amount of time, even only a few minutes, then disconnecting from the server on the Xbox. The server never reappears and phantom has to be restarted for it to show up again.

Suggested fix: detect when a session ends and automatically refresh phantoms connection to the server after a suitable period of time, a minute or two to ensure no false disconnect triggers.

**Comment by azariah001** [comment]
@jhead I'm going to start testing this issue with the worker's flag. Curious to see if it fixes the issue. Will report back in a few days.

**Comment by BRadHoc** [comment]
I'm getting the same issue, the server isn't showing up on the LAN list at all. Updated both the bds and clients to latest version. Can connect to the server via Windows 10 client, but phantom wont ping the console and thus not appearing in the list.

**Comment by azariah001** [comment]
> I'm getting the same issue, the server isn't showing up on the LAN list at all. Updated both the bds and clients to latest version. Can connect to the server via Windows 10 client, but phantom wont ping the console and thus not appearing in the list.

Have you checked that your server is set to broadcast on Lan? If it isn't the Phantom won't show up. Also have you tried manually connecting to the server through Phantom from the Windows 10 client? Using the Phantoms IP and the Port number listed in console when it started.

**Comment by BRadHoc** [comment]
Yes, checked with MCC Tool Chest PE as per the other issue, set to 1. Phantom is reporting an IP of 0.0.0.0 is this normal?

> 5:59PM INF Starting idle connection handler
5:59PM INF Binding ping server to port 19132
5:59PM INF Binding proxy server to: 0.0.0.0:54166
5:59PM INF Listener starting up: 0.0.0.0:19132
5:59PM INF Proxy server listening!
5:59PM INF Once your console pings phantom, you should see replies below.
5:59PM INF Starting 1 workers
5:59PM INF Listener starting up: 0.0.0.0:54166

**Comment by BRadHoc** [comment]
I tried binding the proxy to my internal IP on the network (192.168.1.12)

> brad@Bradleys-MacBook-Pro Phantom % ./phantom-macos -server myserver.com:19132 -debug -bind 192.168.1.12 -bind_port 19132
Starting up with remote server IP: my server.com:19132
6:06PM INF Binding ping server to port 19132
6:06PM INF Starting idle connection handler
6:06PM INF Binding proxy server to: 192.168.1.12:19132
6:06PM INF Proxy server listening!
6:06PM INF Once your console pings phantom, you should see replies below.
6:06PM INF Starting 1 workers
6:06PM INF Listener starting up: 0.0.0.0:19132
6:06PM INF Listener starting up: 192.168.1.12:19132

Minecraft doesn't see the server at 192.168.1.12 or 127.0.0.1 but it see's it and can connect at myserver.com

**Comment by azariah001** [comment]
> Yes, checked with MCC Tool Chest PE as per the other issue, set to 1. Phantom is reporting an IP of 0.0.0.0 is this normal?

Yes, 0.0.0.0 is network speak for universal receiver listening on all interfaces. Is basically a zero-configuration way of ensuring applications providing network services show up as expected by the user on devices with unusual network configurations. Such as laptops that have both WiFi and LAN connections.
